PhysicsTools/Utilities/interface/Primitive.h

```diff
@@ -156,10 +156,8 @@ namespace funct {
   //  /
   
   template <TYPXT2,
-    bool bint = ::boost::type_traits::ice_not<
-    ::boost::is_same<PRIMIT(X, B), UndefinedIntegral>::value>::value,
-    bool aint = ::boost::type_traits::ice_not<
-    ::boost::is_same<PRIMIT(X, A), UndefinedIntegral>::value>::value>
+    bool bint = not ::boost::is_same<PRIMIT(X, B), UndefinedIntegral>::value,
+    bool aint = not ::boost::is_same<PRIMIT(X, A), UndefinedIntegral>::value>
```

Comment:
`std::is_same` would again eliminate the need for boost. Same applies to the stuff bellow.